docs/minio-job.md

```diff
@@ -0,0 +1,208 @@
+      volumeMounts:
+        - name: policy
+          mountPath: /temp
+      volumes:
+        - name: policy
+          configMap:
+            name: mytestconfig
+            items:
+              - key: policy.json
+                path: policy.json
+    - op: admin/policy/attach
+      dependsOn:
+        - add-my-user-1
+        - add-my-user-2
+        - add-my-policy
+      args:
+        policy: memes-access
+        user: daniel
+    - op: admin/policy/attach
+      dependsOn:
+        - add-my-user-1
+        - add-my-user-2
+        - add-my-policy
+      args:
+        policy: memes-access
+        user: pedro
+    - op: stat
+      command:
+        - "mc"
+        - "stat"
+        - "myminio/memes"
+```
+The MinioJob is a Kubernetes Job that runs mc commands. It uses the MinIO client (mc) to interact with the MinIO server.
+## mcImage
+The `mcImage` field specifies the Docker image that will be used to run the mc commands. In this case, the image is `minio/mc:latest`.
+## serviceAccountName
+The `serviceAccountName` field specifies the name of the Kubernetes ServiceAccount that will be used to run the mc commands. In this case, the ServiceAccount is `mc-job-sa`.
+## securityContext
+example:
+```yaml
+runAsUser: 1000
+runAsGroup: 1000
+fsGroup: 1000
+fsGroupChangePolicy: "OnRootMismatch"
+runAsNonRoot: true
+allowPrivilegeEscalation: false
+capabilities:
+  drop:
+    - ALL
+```
+The `securityContext` field specifies the security context that will be used to run the mc commands. 
+## containerSecurityContext
+The `containerSecurityContext` field specifies the security context that will be used to run the `mc` commands in the container.
+## tenant
+```yaml
+name: tenantName
+namespace: tenantNamespace
+```
+The target tenant that the job will run against.
+## commands
+### args
+if you set this field, the `mc` command will be executed with the arguments.
+`op` must be one of these:
+`make-bucket`, `admin/user/add`,`admin/policy/create`,`admin/policy/attach`, `admin/config/set`, `support/callhome`,`license/register`
+```yaml
+op: make-bucket
+args:
+  name: memes
+```
+Will do a job like `mc mb myminio/memes`
+```yaml
+name: add-my-policy
+op: admin/policy/create
+args:
+  name: memes-access
+policy: /temp/policy.json
+volumeMounts:
+- name: policy
+  mountPath: /temp
+volumes:
+- name: policy
+  configMap:
+    name: mytestconfig
+    items:
+      - key: policy.json
+        path: policy.json
+```
+Will do a job like `mc admin policy create myminio memes-access /temp/policy.json`
+### command
+The `command` field specifies the command that will be executed by the `mc` command.
+`args` must be empty. And `op` can be set to the main command name.
+```
+op: stat
+command:
+  - "mc"
+  - "stat"
+  - "myminio/memes"
+```
+Will do a job like `mc stat myminio/memes`
+### env/envFrom/volumeMounts/volumes
+The `env/envFrom/volumeMounts/volumes` fields specify the environment variables/volumes that will be used by the `mc` command
```

Comment:
Are these for the commands, or a way to pass envvars to MC itself?

Reply:
We may want to specify which shell is being used to run the command, so users know what they can use on the commandline.